docs/madr/decisions/076-migrating-to-kri-based-envoy-resource-and-stat-naming.md

```diff
@@ -0,0 +1,260 @@
+```
+
+#### ZoneIngress and ZoneEgress proxies
+
+`ZoneIngress` and `ZoneEgress` proxies must also include the KRI naming feature flag to ensure consistent naming across the mesh. In both Universal and Kubernetes modes, this is done by setting the following environment variable in their deployments:
+
+```env
+KUMA_DATAPLANE_RUNTIME_KRI_NAMING_ENABLED=true
+```
+
+#### Helper setting for Kubernetes installations
+
+To simplify configuration in Kubernetes environments, a new installation setting will be introduced:
+
+```yaml
+experimental.kriNaming.enabled
+```
+
+When set to `true`, this will:
+
+* Set the `KUMA_DATAPLANE_RUNTIME_KRI_NAMING_ENABLED="true"` environment variable in all `ZoneIngress` and `ZoneEgress` deployments
+* Set the `KUMA_RUNTIME_KUBERNETES_INJECTOR_KRI_NAMING_ENABLED="true"` environment variable in the control plane deployment
+
+Once KRI naming becomes the default, this setting will also default to `true`. Users will still be able to disable it by setting it to `false`. Eventually, this setting will be removed when disabling the feature is no longer supported.
+
+### Naming for inbound-related resources
+
+The current KRI naming format, as defined in the [Resource Identifier](070-resource-identifier.md) MADR, proposes that non-internal inbound resources (such as listeners and clusters) use the full KRI name of the originating `Dataplane` plus the inbound port/name as the `sectionName`. This would result in names like:
+
+```
+kri_dp_default_kuma-2_kuma-demo_demo-app-ddd8546d5-vg5ql_5050
+```
+
+While technically correct and traceable, this approach significantly increases metrics cardinality compared to the current convention (e.g., `localhost_5050`), and would cause noticeable regressions in performance, cost, and compatibility with existing observability tooling. To mitigate this, we are considering alternative naming schemes for these inbound-related resources:
+
+#### Option 1: Replace dataplane name with a simple keyword (`self` or `this`)
+
+Use a naming format like `{prefix}_{sectionName}`, where prefix is a static term such as `self` or `this`. Example:
+
+```
+self_5050
+```
+
+This would apply to both resource names and stat names for non-internal inbounds.
+
+**Benefits:**
+
+* No increase in metric cardinality
+* Minimal changes to existing tooling: `localhost_*` just becomes `self_*` or `this_*`
+
+**Drawbacks:**
+
+* Breaks the original Resource Identifier model by introducing a third category of resources (not Kuma-based, not internal, but still service-to-service and not uniquely identified)
+* Reduces consistency in the overall naming convention
+
+**Neutral:**
+
+* Provides no more or less value than the current `localhost_*` naming scheme
+
+#### Option 2: Treat inbounds as "restricted" and use `_kuma_{prefix}_{sectionName}` format
+
+Update the [Internal listeners naming convention](036-internal-listeners.md) and [Resource Identifier](070-resource-identifier.md) MADRs by:
```

Comment:
My MADR https://github.com/kumahq/kuma/pull/13783/files supersedes that one, but it needs to land

```suggestion
Update the [Internal resources naming convention](074-naming-internal-envoy-resources.md) and [Resource Identifier](070-resource-identifier.md) MADRs by:
```